src/compile/mark/valueref.ts

```diff
@@ -216,13 +225,22 @@ export function tooltipForChannelDefs(channelDefs: FieldDef<string>[], config: C
   return keyValues.length ? {signal: `{${keyValues.join(', ')}}`} : undefined;
 }
 
-export function text(textDef: ChannelDefWithCondition<TextFieldDef<string>>, config: Config): VgValueRef {
+export function text(channelDef: ChannelDefWithCondition<TextFieldDef<string>>, config: Config): VgValueRef {
   // text
-  if (textDef) {
-    if (isFieldDef(textDef)) {
-      return formatSignalRef(textDef, textDef.format, 'datum', config);
-    } else if (isValueDef(textDef)) {
-      return {value: textDef.value};
+  if (channelDef) {
+    if (isValueDef(channelDef)) {
+      return {value: channelDef.value};
+    }
+    let format;
```

Comment:
Extract a format function? 
(And check all references of `format`, so we can reuse it and fix similar problem elsewhere?)